plaso/containers/artifacts.py

```diff
@@ -66,6 +67,120 @@ def __init__(self, name=None, schema='DNS'):
     self.schema = schema
 
 
+class OperatingSystemArtifact(ArtifactAttributeContainer):
+  """Operating system artifact attribute container.
+
+  Attributes:
+    name (str): name, such as "Linux", "MacOS" or "Windows".
+    product (str): product name, such as "macOS Mojave" or "Windows XP".
+    version (str): version, such as "10.14.1" or "5.1".
+  """
+  CONTAINER_TYPE = 'operating_system'
+
+  # Note that the product names below are normalized.
+  _NAMES_AND_VERSIONS_PER_PRODUCT = {
+      'Windows 2000': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 0)),
+      'Windows 2003': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 2)),
+      'Windows 2003 R2': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 2)),
+      'Windows 2008': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 0)),
+      'Windows 2008 R2': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 1)),
+      'Windows 2012': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 2)),
+      'Windows 2012 R2': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 3)),
+      'Windows 2016': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (10, 0)),
+      'Windows 2019': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (10, 0)),
+      'Windows 7': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 1)),
+      'Windows 8': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 2)),
+      'Windows 8.1': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 3)),
+      'Windows 10': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (10, 0)),
+      'Windows Vista': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (6, 0)),
+      'Windows XP': (definitions.OPERATING_SYSTEM_WINDOWS_NT, (5, 1))}
+
+  def __init__(self, name=None, product=None, version=None):
+    """Initializes an operating system artifact.
+
+    Args:
+      name (str): name, such as "Linux", "MacOS" or "Windows".
+      product (str): product name, such as "macOS Mojave" or "Windows XP".
+      version (str): version, such as "10.14.1" or "5.1".
+    """
+    super(OperatingSystemArtifact, self).__init__()
+    self.name = name
+    self.product = product
+    self.version = version
+
+  @property
+  def normalized_product(self):
+    """str: normalized product name."""
+    normalized_product = self.product or ''
+    normalized_product = normalized_product.split(' ')
+    normalized_product_lower_case = [
+        segment.lower() for segment in normalized_product]
+
+    if 'windows' in normalized_product_lower_case:
+      segment_index = normalized_product_lower_case.index('windows') + 1
+      if normalized_product_lower_case[segment_index] == 'server':
+        segment_index += 1
+      return 'Windows {0:s}'.format(normalized_product[segment_index])
+
+    return self.product or ''
+
+  @property
+  def version_tuple(self):
+    """tuple[int]: version tuple or None if version is not set or invalid."""
+    try:
+      return tuple([int(digit, 10) for digit in self.version.split('.')])
+    except (AttributeError, TypeError, ValueError):
+      return None
+
+  def Compare(self, other):
```

Comment:
Please rename to "Equal" or "IsEquivalent" or something. "Compare" is too generic for what this method does. It's not clear what sort of comparison is being performed.

Reply:
ack.